llvm/lib/Target/AMDGPU/SIMachineFunctionInfo.cpp

```diff
@@ -344,6 +342,8 @@ void SIMachineFunctionInfo::shiftSpillPhysVGPRsToLowestRange(
       MBB.removeLiveIn(Reg);
       MBB.sortUniqueLiveIns();
     }
+
+    Reg = NewReg;
```

Comment:
I find this a bit subtle. It updates the entry in SpillPhysVGPRs. I don't know if there's a good way to make this more obvious to the reader.